static/js/components/EducationTab.js

```diff
@@ -0,0 +1,198 @@
+
+class EducationTab extends ProfileTab {
+  constructor(props) {
+    super(props);
+    this.boundEducationMonthField = boundEducationMonthField.bind(this);
+    this.boundEducationYearField = boundEducationYearField.bind(this);
+    this.educationLevelOptions = [
+      { value: 'hs', label: "High school", enabled: true },
+      { value: 'a', label: 'Associate degree', enabled: true },
+      { value: 'b', label: "Bachelor's degree", enabled: true },
+      { value: 'm', label: "Master's or professional degree", enabled: false },
+      { value: 'p', label: "Doctorate", enabled: false }
+    ];
+    this.educationLevelLabels = {
+      'hs' : "High school",
+      'a': 'Associate degree',
+      'b': "Bachelor's degree",
+      'm': "Master's or professional degree",
+      'p': "Doctorate"
+    };
+
+    this.saveAndContinue = saveAndContinue.bind(this, '/dashboard');
+    this.handleOpenDialog = this.handleOpenDialog.bind(this);
+    this.handleCloseDialog = this.handleCloseDialog.bind(this);
+  }
+  printAddDegree(level){
+    const { educationLevels } = this.props;
+    if (educationLevels[level.value]){
+      return <Grid key={"add-"+level.value}>
+        <Cell col={11}></Cell>
+        <Cell col={1}>
+          <FABButton mini onClick={this.handleOpenDialog.bind(this, level.value, null )} raised ripple>
+            <Icon name="add" />
+          </FABButton>
+        </Cell>
+
+      </Grid>;
+    }
+  }
+  printExistingEducations(level){
+    const { profile } = this.props;
+    return profile['education'].map(education => {
+      if(education.degree_name === level.value && 'id' in education) {
+        return <Grid className="existing-education-grid">
+          <Cell col={3}>{this.educationLevelLabels[education.degree_name]}</Cell>
+          <Cell col={7}>{education.graduation_date}</Cell>
+          <Cell col={1}>
+            <Button ><Icon name="edit"/></Button>
+          </Cell>
+          <Cell col={1}>
+            <Button ><Icon name="delete"/></Button>
+          </Cell>
+        </Grid>;
+      }
+    });
+  }
+  saveEducationForm(){
+    const { dispatch, saveProfile, profile } = this.props;
+    saveProfile(profile).then(() => {
+      dispatch(closeEducationForm());
+    });
+
+  }
+  handleSwitchClick(level){
+    const {dispatch} = this.props;
+    dispatch(toggleEducationLevel(level))
+
+  }
+  handleOpenDialog(level, index){
+    const { dispatch, profile, updateProfile } = this.props;
+    let new_index = index;
+    if (index === null){
+      new_index = profile['education'].length;
+    }
+    dispatch(openEducationForm(level, new_index));
+
+    let clone = Object.assign({}, profile);
+    clone['education'].push(this.addNewEducation(level));
+    updateProfile(clone);
+
+  }
+  handleCloseDialog(){
+    const { dispatch } = this.props;
+    dispatch(closeEducationForm())
+  }
+  addNewEducation(level){
+    return {
+      'degree_name': level,
```

Comment:
Can we move this function into `util.js` so we can unit test it? There's a similar object in the employment tab which we should move into a function also